Panic on windows when accepting new job

This is with buildkite-agent 3.15.0 when running on Windows Server 2019. I've reverted back to 3.14.0 which works fine. The backtrace is:
```
...
Assigned job 0baca7fb-f1c5-4569-8120-6b52747899c0. Accepting...
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x75cfad]

goroutine 104 [running]:
github.com/buildkite/agent/process.(*Process).Done(0x0, 0x0)
	/go/src/github.com/buildkite/agent/process/process.go:268 +0x2d
github.com/buildkite/agent/agent.NewJobRunner.func3(0xc0000ea1a0, 0xc0003a8040, 0x145d660, 0xc00008a9c0)
	/go/src/github.com/buildkite/agent/agent/job_runner.go:180 +0x3d
created by github.com/buildkite/agent/agent.NewJobRunner
	/go/src/github.com/buildkite/agent/agent/job_runner.go:179 +0x8d3
```

--- a/agent/job_runner.go
+++ b/agent/job_runner.go
@@ -175,14 +175,6 @@ func NewJobRunner(l logger.Logger, scope *metrics.Scope, ag *api.AgentRegisterRe
 		}()
 	}
 
-	// Close the writer end of the pipe when the process finishes
-	go func() {
-		<-runner.process.Done()
-		if err := pw.Close(); err != nil {
-			l.Error("%v", err)
-		}
-	}()
-
 	// Copy the current processes ENV and merge in the new ones. We do this
 	// so the sub process gets PATH and stuff. We merge our path in over
 	// the top of the current one so the ENV from Buildkite and the agent
@@ -199,6 +191,14 @@ func NewJobRunner(l logger.Logger, scope *metrics.Scope, ag *api.AgentRegisterRe
 		Stderr:          processWriter,
 		InterruptSignal: conf.CancelSignal,
 	})
+
+	// Close the writer end of the pipe when the process finishes
+	go func() {
+		<-runner.process.Done()
+		if err := pw.Close(); err != nil {
+			l.Error("%v", err)
+		}
+	}()
 
 	// Kick off our callback when the process starts
 	go func() {

